packages/core/src/types.ts

```diff
@@ -324,3 +324,21 @@ export type RequestedEntitiesMessage = {
   entities: Array<Entry | Asset>;
   missingEntityIds?: string[];
 };
+
+//All the possible types that can be returned to a component prop
+export type BoundComponentPropertyTypes =
+  | string
+  | number
+  | boolean
+  | AssetFile
+  | Record<string, AssetFile | undefined>
+  | Document
+  | OptimizedImageAsset
+  | undefined;
+
+export type OptimizedImageAsset = {
+  url: string;
+  srcSet?: string[];
+  sizes?: string;
```

Comment:
shouldn't `sizes` also be an array - accounting for different breakpoints - that we join?

Reply:
I was thinking that whenever possible we should be utilizing the inherent responsiveness of HTML and CSS, as they are essentially breakpoint-aware by design (which can't be said about React props).

This goes somewhat hand-in-hand with the best possible support for SSR. If `sizes` change dynamically upon hydration, we're not leveraging the above to the fullest.

Reply:
I think there is some confusion on what the `sizes` variable here does. End the end, it populates the `sizes` attribute on the `img` tag. Users set this value on the design tab, and they are able to control which value gets passed in via our already existing breakpoint system. So, for instance, they can set different sizes for each of our breakpoints (desktop, tablet, and mobile).

Reply:
Yes, I think I have a pretty good understanding of it.
My point is simply that whenever possible we should utilize media queries and create static HTML (provide a wholistic image tag `sizes` attribute value that doesn't have to be updated), rather than rely on stateful React (update `sizes` depending on the current screen size).

The `sizes` **variable** would function the same way - like you said, it uses the breakpoint system - but when actualy rendering via React, we would take all the breakpoint values and generate a single `sizes` combining provided sizes for all breakpoints for the image tag.

Anyways, probably not super important and something we could optimize later still.